gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -380,6 +439,8 @@ gis_install_page_copy (GisPage *page)
 
   g_thread_yield();
 
+  priv->writing = TRUE;
```

Comment:
There's a potential race here, where you click the Close button, and then the main thread happens to yield, and only then is this thread scheduled, and it runs for long enough to overwrite a little bit of the disk.

Could add a mutex and a GCancellable (say); have the `delete` callback claim the mutex, check `writing`; if it is false, then cancel the GCancellable, and only then release the mutex. Here, you'd claim the mutex, check the cancellable, set `writing` and then release the mutex.